tests/test_cookies.py

```diff
@@ -101,8 +101,52 @@ def test_bake_project(cookies):
 
             assert result.exit_code == 0
             assert result.exception is None
+
+            assert result.project_path.name == 'helloworld'
+            assert result.project_path.is_dir()
+            assert str(result) == '<Result {}>'.format(result.project_path)
+
             assert result.project.basename == 'helloworld'
             assert result.project.isdir()
+            assert str(result) == '<Result {}>'.format(result.project)
+    """
+    )
+
+    result = testdir.runpytest("-v", "--template={}".format(cookiecutter_template))
+
+    result.stdout.fnmatch_lines(["*::test_bake_project PASSED*"])
+
+
+def test_cookies_bake_project_warning(testdir, cookiecutter_template):
+    """Programmatically create a **Cookiecutter** template and use `bake` to
+    create a project from it and check for warnings when accesssing the project
+    attribute.
+    """
+    testdir.makepyfile(
+        """
+        # -*- coding: utf-8 -*-
+        import warnings
+
+        def test_bake_project(cookies):
+            warning_message = (
+                "project is deprecated and will be removed in a future release, "
+                "please use project_path instead."
+            )
+
+            result = cookies.bake(extra_context={'repo_name': 'helloworld'})
+
+            assert result.exit_code == 0
+            assert result.exception is None
+
+            with warnings.catch_warnings(record=True) as w:
+                warnings.simplefilter("always")
+                project_basename = result.project.basename
+
+            assert project_basename == 'helloworld'
+
+            [warning] = w
```

Comment:
🤔 presumably this is the same as `warning = w[0]`?

Reply:
This will result in `ValueError: too many values to unpack (expected 1)`:
```python
warnings = ["hello", "world", "pytest-cookies"]
[w] = warnings
```

This will store the only item in the `warnings` list in `w`:
```python
warnings = ["project deprecated"]
[w] = warnings
```